scripts/generate-apt-packages-list.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -e
+
+# This script generates a list of packages with their versions and other details to make it easier for parsing by check-repository-consistency.js
+# Outputs scripts/apt-packages-list-<arch>.txt for each architecture
+#
+# Format of each line:
+# <package_name> <repo-name> <version> <may_have_staticsplit>
+
+TERMUX_PACKAGES_DIR="$(realpath "$(dirname "$(realpath "$0")")/..")"
+repo_paths=$(jq --raw-output 'del(.pkg_format) | keys | .[]' "$TERMUX_PACKAGES_DIR/repo.json")
+
+for arch in "aarch64" "arm" "i686" "x86_64"; do
+	for repo_path in ${repo_paths[@]}; do
+		repo_name=$(jq --raw-output ".\"$repo_path\".name" "$TERMUX_PACKAGES_DIR/repo.json")
+		package_paths=($TERMUX_PACKAGES_DIR/$repo_path/*)
+		for pkg_path in "${package_paths[@]}"; do
+			(
+				. "$pkg_path/build.sh" &> /dev/null || :
+				# We are erraneously adding a revision for packages with version containing a dash when generating the apt packages in our build scripts.
+				# So reproduce the same bug here
+        # Source: scripts/build/termux_extract_dep_info.sh#18-20
+				if [[ "${TERMUX_PKG_VERSION/-/}" != "${TERMUX_PKG_VERSION}" ]] && [[ "$TERMUX_PKG_REVISION" == "" ]]; then
+					export TERMUX_PKG_REVISION=0
+				fi
+				IFS="," read -r -a EXCLUDED_ARCHES <<< "${TERMUX_PKG_EXCLUDED_ARCHES:-}"
+				excluded=false
+				for excluded_arch in ${EXCLUDED_ARCHES[@]}; do
+					if [[ "$excluded_arch" = "$arch" ]]; then
+						excluded=true
+					fi
+				done
+				if [[ "$excluded" != true ]]; then
+					if [[ -d "$pkg_path" ]]; then
+						if [[ "$TERMUX_PKG_REVISION" != "" ]]; then
+							echo -n "$(basename "$pkg_path") $repo_name $TERMUX_PKG_VERSION-$TERMUX_PKG_REVISION"
+						else
+							echo -n "$(basename "$pkg_path") $repo_name $TERMUX_PKG_VERSION"
+						fi
+					fi
+					if [[ "$TERMUX_PKG_NO_STATICSPLIT" = true ]]; then
+						echo " false"
+					else
+						echo " true"
+					fi
+					for subpkg in "$pkg_path"/*.subpackage.sh; do
+						if [[ -f "$subpkg" ]]; then
+							(
+								. "$subpkg" &> /dev/null || :
+								IFS="," read -r -a SUBPKG_EXCLUDED_ARCHES <<< "${TERMUX_SUBPKG_EXCLUDED_ARCHES:-}"
+								excluded=false
+								for excluded_subpkg_arch in ${SUBPKG_EXCLUDED_ARCHES[@]}; do
+									if [[ "$excluded_subpkg_arch" = "$arch" ]]; then
+										excluded=true
+									fi
+								done
+								if [[ "$excluded" != true ]]; then
+									if [[ "$TERMUX_PKG_REVISION" != "" ]]; then
+										echo "$(basename "$(basename "$subpkg" .subpackage.sh)") $repo_name $TERMUX_PKG_VERSION-$TERMUX_PKG_REVISION false"
+									else
+										echo "$(basename "$(basename "$subpkg" .subpackage.sh)") $repo_name $TERMUX_PKG_VERSION false"
+									fi
+								fi
+							)
+						fi
+					done
+				fi
+			)
+		done
+	done > "$TERMUX_PACKAGES_DIR/scripts/apt-packages-list-$arch.txt" & # Parallelize each architecture
```

Comment:
This just reads a few files and does a trivial amount of work on each, right? What are we actually gaining by introducing parallelism? At least the other script is doing some network requests. (I wouldn't parallelize that, either.)

Reply:
It gives significant boost when testing locally from 20s to 5s locally, I think it's worth it, considering it just works without having to do anything

Reply:
Why are we putting this into `scripts/` anyway?
These digests should go into `$TERMUX_PKG_TMPDIR`.
I see absolutely no reason to put them ***in*** the repo.

Reply:
Yeah that sounds fine.
We might wanna use `${TMPDIR:-/tmp}` for on-device compatibility in Termux.

Reply:
This isn't supposed to be run on device, but just as a github cron job to notify us periodically about if anything is wrong with any apt repository

Reply:
I'll go ahead with a note. Maybe premature optimization, but I don't think it's much of a deal, anyways this script isn't used (currently) for anything else rather than repository health check, so I think it's fine.

Reply:
Those 20 seconds do add up pretty quick when debugging locally. I would still prefer to go with keeping the parallelism